src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -362,6 +316,236 @@ protected override void StopProcessing()
         }
     }
 
+    /// <summary>
+    /// Implementation for the ConvertTo-Clixml command.
+    /// </summary>
+    [Cmdlet(VerbsData.ConvertTo, "Clixml", SupportsShouldProcess = false,
+        RemotingCapability = RemotingCapability.None)]
+    [OutputType(typeof(XmlDocument), typeof(string))]
+    public sealed class ConvertToClixmlCommand : PSCmdlet, IDisposable
+    {
+        #region Command Line Parameters
+
+        /// <summary>
+        /// Gets or sets depth of serialization.
+        /// </summary>
+        [Parameter(HelpMessage = "Specifies how many levels of contained objects should be included in the XML representation")]
+        [ValidateRange(1, int.MaxValue)]
+        public int Depth { get; set; } = 0;
+
+        /// <summary>
+        /// Input Object which is written to XML format.
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipeline = true, Mandatory = true)]
+        [AllowNull]
+        public PSObject InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the As parameter.
+        /// </summary>
+        [Parameter]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("String", "Document")]
+        public string As { get; set; } = "Document";
+
+        #endregion Command Line Parameters
+
+        #region Overrides
+
+        /// <summary>
+        /// BeginProcessing override.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            CreateMemoryStream();
+        }
+
+        /// <summary>
+        /// Override ProcessRecord.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            if (_serializer != null)
```

Comment:
It looks like _serializer is always created in 'CreateMemoryStream', so there is no need to check if it is null.

Reply:
Yeah, you're right, but it should also be moved completely back to the `ExportXmlHelper` class. 

The idea here is that `ConvertTo/Export-Clixml` and `ConvertFrom/Import-CliXml` become thin wrappers around the helpers which in turn abstracts away the the (de-)serialization interfaces, and the cmdlets themselves just feed a reader or writer to the helper class.